infra/embedding_step_functions/unified_embedding/handlers/compaction.py

```
@@ -1188,49 +1357,72 @@ def final_merge_handler(event: Dict[str, Any]) -> Dict[str, Any]:
 
-    logger.info(
-        "Pre-downloading intermediates before lock acquisition",
-        intermediate_count=len(intermediate_keys_to_download),
-        batch_id=batch_id,
-    )
+    if precompacted:
+        # Use the pre-compacted intermediate directly - skip S3 download entirely
+        synthetic_key = precompacted["key"]
+        temp_dir = precompacted["temp_dir"]
+        predownloaded_intermediates.append((synthetic_key, temp_dir))
+        owns_precompacted_dir = True  # We're responsible for cleanup
```

Ruff (0.14.10)

Local variable `owns_precompacted_dir` is assigned to but never used

Remove assignment to unused variable `owns_precompacted_dir`

(F841)